outputs.tf

```diff
@@ -0,0 +1,27 @@
+output "detector" {
+  description = "GuardDuty filter"
+  value       = aws_guardduty_detector.this
+}
+output "filter" {
+  description = "GuardDuty filter"
+  value       = var.filter != null ? aws_guardduty_filter.this : null
+}
+
+output "ipset" {
+  description = "GuardDuty ipset"
+  value       = var.ipset != null ? aws_guardduty_ipset.this : null
+}
+
+output "threatintelset" {
+  description = "GuardDuty threatintelset"
+  value       = var.threatintelset != null ? aws_guardduty_threatintelset.this : null
+}
+
+output "publishing_destination" {
+  description = "GuardDuty publishing destination"
+  value       = var.publishing_destination != null ? aws_guardduty_publishing_destination.this : null
```

Comment:
```suggestion
  value       = aws_guardduty_publishing_destination.this
```

Reply:
Non-concur since the default for this variable is null and the user does not have to include one.

Reply:
Give it a try. It doesn't have anything to do with the default. When the resource is not created due to the count/for_each condition, the output will be `null`. The extra logic is just complicating terraform's standard behavior.